.github/workflows/java.yml

```diff
@@ -313,6 +322,21 @@ jobs:
                   version: "29.1"
                   repo-token: ${{ secrets.GITHUB_TOKEN }}
 
+            # Ensure Rust is in PATH for container environments
+            - name: Setup Rust Build
+              if: ${{ contains(matrix.host.TARGET, 'musl') }}
+              run: |
+                  echo 'source "$HOME/.cargo/env"' >> ~/.bashrc
+                  source "$HOME/.cargo/env"
+                  echo "PATH=$HOME/.cargo/bin:$PATH" >> $GITHUB_ENV
+                  # Verify Rust is available
+                  which cargo || echo "Cargo still not found in PATH"
+                  echo "Current PATH: $PATH"
+
+                  # Install ziglang and zigbuild
+                  pip3 install ziglang --break-system-packages
+                  cargo install --locked cargo-zigbuild
```

Comment:
You don't need zigbuild on musl. It is used to ensure compatibility with older version of GLIBC on GNU platforms.
So you can probably revert changes in `.github/workflows/install-shared-dependencies/action.yml`